compiler/rustc_mir/src/transform/check_consts/validation.rs

```diff
@@ -425,18 +425,20 @@ impl Visitor<'tcx> for Validator<'mir, 'tcx> {
 
         match elem {
             ProjectionElem::Deref => {
-                let base_ty = Place::ty_from(place_local, proj_base, self.body, self.tcx).ty;
-                if let ty::RawPtr(_) = base_ty.kind() {
-                    if proj_base.is_empty() {
-                        if let (local, []) = (place_local, proj_base) {
-                            let decl = &self.body.local_decls[local];
-                            if let Some(box LocalInfo::StaticRef { def_id, .. }) = decl.local_info {
-                                let span = decl.source_info.span;
-                                self.check_static(def_id, span);
-                                return;
-                            }
+                if proj_base.is_empty() {
+                    if let (local, []) = (place_local, proj_base) {
```

Comment:
What's up with this?

Reply:
This used to be handcrafted to the raw pointers used for mutable statics, now that they actually use references, we need to stop checking for raw pointers. Everything else still applies

/me goes write comments on this thing, because I just stared at it for minutes to figure out what it does

Reply:
I think you only need the second I mean.